eventuate-tram-consumer-activemq/src/main/java/io/eventuate/tram/consumer/activemq/MessageConsumerActiveMQImpl.java

```diff
@@ -42,12 +42,27 @@
   private AtomicBoolean runFlag = new AtomicBoolean(true);
 
   public MessageConsumerActiveMQImpl(String url) {
-    this(url, Collections.emptyMap());
+    this(url, Collections.emptyMap(), Optional.empty(), Optional.empty());
   }
 
-  public MessageConsumerActiveMQImpl(String url, Map<String, ChannelType> messageModes) {
+  public MessageConsumerActiveMQImpl(String url,
```

Comment:
Why have more than one constructor? It makes the class confusing. If test code needs multiple constructors then define in src/test a class with factory methods

Reply:
I will fix